projects/from-javascript-to-typescript/the-typeinator/02-prototypes-to-classes/index.test.js

```diff
@@ -0,0 +1,162 @@
+const index = require("./index");
+const solution = require("./solution");
+
+// Use solution when TEST_SOLUTIONS is true, otherwise try to use index (may be empty)
+const { Robot, Humanoid } = process.env.TEST_SOLUTIONS ? solution : index || {};
+
+describe("02 - Prototypes to Classes", () => {
+	// Skip tests if Robot/Humanoid classes are not implemented
+	const shouldSkip = !Robot || !Humanoid;
+
+	describe("Robot", () => {
+		(shouldSkip ? test.skip : test)(
+			"creates a robot with name, abilities, and power",
+			() => {
+				const robot = new Robot("R2-D2", ["beep", "navigate"]);
+
+				expect(robot.name).toBe("R2-D2");
+				expect(robot.abilities).toEqual(["beep", "navigate"]);
+				expect(robot.power).toBe(100);
+			},
+		);
+
+		(shouldSkip ? test.skip : test)(
+			"announce method logs greetings and abilities",
+			() => {
+				const consoleSpy = jest.spyOn(console, "log").mockImplementation();
+				const robot = new Robot("C-3PO", ["translate", "worry"]);
+
+				robot.announce();
+
+				expect(consoleSpy).toHaveBeenCalledWith("Greetings. I am C-3PO.");
+				expect(consoleSpy).toHaveBeenCalledWith("I am able to translate.");
+				expect(consoleSpy).toHaveBeenCalledWith("I am able to worry.");
+
+				consoleSpy.mockRestore();
```

Comment:
[Bug] Putting restores after test assertions means they won't run if any assertion throws. That'll result in buggy behavior. Also, every test here is doing the same spy behavior on the console. 

Change request: use [Jest setup and teardown](https://jestjs.io/docs/setup-teardown) APIs for console spies.